qiskit/ignis/verification/accreditation/qotp.py

```diff
@@ -0,0 +1,250 @@
+    twoqubitgate = layers['twoqubitgate']
+    qotp_circ = QuantumCircuit(qregs, cregs)
+    tempCirc = QuantumCircuit(qregs, cregs)
+
+    # initial z gates after prep
+    paulizs = np.random.randint(2, size=len(qregs))
+    for qind, q in enumerate(qregs):
+        if paulizs[qind]:
+            tempCirc.z(q)
+    # step through layers
+    for lnum, gates2q in enumerate(layers['twoqubitlayers']):
+        # add single qubit gates to temp circuit
+        tempCirc = tempCirc+layers['singlequbitlayers'][lnum]
+        # generate and add single qubit paulis
+        paulizs = np.random.randint(2, size=len(qregs))
+        paulixs = np.random.randint(2, size=len(qregs))
+        for qind, q in enumerate(qregs):
+            if paulizs[qind]:
+                tempCirc.z(q)
+            if paulixs[qind]:
+                tempCirc.x(q)
+        # add to circuit and reset temp
+        tempCirc = transpile(tempCirc,
+                             basis_gates=['u1', 'u2', 'u3'])
+        qotp_circ = qotp_circ+tempCirc
+        tempCirc = QuantumCircuit(qregs, cregs)
+        # add two qubit layers and get indices for 2qgates
+        qotp_circ.barrier()
+        qotp_circ = qotp_circ+gates2q
+        qotp_circ.barrier()
+        twoqindices = []
+        for _, qsub, _ in gates2q:
+            twoqindices.append([qsub[0].index, qsub[1].index])
+        # update Paulis
+        for inds in twoqindices:
+            if twoqubitgate == 'cx':
+                # iz -> zz and xi -> xx
+                paulizs[inds[0]] = (paulizs[inds[0]]+paulizs[inds[1]]) % 2
+                paulixs[inds[1]] = (paulixs[inds[1]]+paulixs[inds[0]]) % 2
+            elif twoqubitgate == 'cz':
+                # ix -> zx and xi -> xz
+                paulizs[inds[0]] = (paulizs[inds[0]]+paulixs[inds[1]]) % 2
+                paulizs[inds[1]] = (paulizs[inds[1]]+paulixs[inds[0]]) % 2
+            else:
+                raise Exception("Two qubit gate {0}".format(twoqubitgate)
+                                + "is not implemented in qotp")
+        for qind, q in enumerate(qregs):
+            if paulixs[qind]:
+                tempCirc.x(q)
+            if paulizs[qind]:
+                tempCirc.z(q)
+    # add final single qubit layer
+    tempCirc = tempCirc+layers['singlequbitlayers'][-1]
+    # add final Paulis to create the one time pad
+    paulizs = np.random.randint(2, size=len(qregs))
+    paulixs = np.random.randint(2, size=len(qregs))
+    for qind, q in enumerate(qregs):
+        if paulizs[qind]:
+            tempCirc.z(q)
+        if paulixs[qind]:
+            tempCirc.x(q)
+    # add to circuit
+    tempCirc = transpile(tempCirc,
+                         basis_gates=['u1', 'u2', 'u3'])
+    qotp_circ = qotp_circ+tempCirc
+    # post operations
+    qotp_postp = np.flip(paulixs)
+    # measurements
+    qotp_circ = qotp_circ+layers['measlayer']
+    return qotp_circ, qotp_postp
+
+
+def QOTP(circ, num, twoqubitgate='cx', coupling_map=None):
```

Comment:
```suggestion
def QOTP(circ, num, two_qubit_gate='cx', coupling_map=None):
```

Different default value to the other file